src/test/java/ca/utoronto/fitbook/integration/UserProfileControllerIntegrationTest.java

```diff
@@ -45,15 +51,15 @@ public void cleanUp() {
 
     @Test
     public void findExistingUserReturnsProfileWithValidContentAndAttributesTest() throws Exception {
-        this.mockMvc.perform(get(String.format("/profile/%s", testUser.getId())))
+        this.mockMvc.perform(get(String.format("/profile/%s", testUser.getId())).session(session))
                 .andExpect(status().isOk())
                 .andExpect(content().contentType("text/html;charset=UTF-8"))
                 .andExpect(model().attributeExists("profile"));
     }
 
     @Test
     public void findNonExistentUserProfileReturnsServerErrorHttpStatusTest() throws Exception {
-        this.mockMvc.perform(get("/profile/-1"))
+        this.mockMvc.perform(get("/profile/-1").session(session))
                 .andExpect(status().is5xxServerError());
```

Comment:
```suggestion
                .andExpect(status().is4xxServerError());
```
I believe we discussed that it should 400 lever error

Reply:
Do you mean a 404 error? Or another kind of error?

Reply:
Oops, my bad I should've been more specific. Make it a 422 error as that is specific to what goes wrong. Thank you.

Reply:
No, thank you! That makes sense, the error we throw should be more informative about what went wrong. I will make these changes!

Reply:
Great! I appreciate your effort! Good Job!